Userland/Libraries/LibAudio/FlacLoader.h

```diff
@@ -102,32 +99,31 @@ class FlacLoaderPlugin : public LoaderPlugin {
     bool sample_count_unknown() const { return m_total_samples == 0; }
 
 private:
-    bool parse_header();
+    MaybeLoaderError parse_header();
     // Either returns the metadata block or sets error message.
     // Additionally, increments m_data_start_location past the read meta block.
-    FlacRawMetadataBlock next_meta_block(InputBitStream& bit_input);
+    Result<FlacRawMetadataBlock, LoaderError> next_meta_block(InputBitStream& bit_input);
     // Fetches and sets the next FLAC frame
-    void next_frame();
+    MaybeLoaderError next_frame();
     // Helper of next_frame that fetches a sub frame's header
-    FlacSubframeHeader next_subframe_header(InputBitStream& bit_input, u8 channel_index);
+    Result<FlacSubframeHeader, LoaderError> next_subframe_header(InputBitStream& bit_input, u8 channel_index);
```

Comment:
I think `Result<T, E>` is mostly deprecated at this point, please use `ErrorOr<T, E>` (with `E` inheriting from `Error`)

Reply:
Yeah, this work is older than ErrorOr...